src/transformers/modeling_utils.py

```diff
@@ -1160,12 +1160,13 @@ def num_parameters(self, only_trainable: bool = False, exclude_embeddings: bool
                 # For 4bit models, we need to multiply the number of parameters by 2 as half of the parameters are
                 # used for the 4bit quantization (uint8 tensors are stored)
                 if is_loaded_in_4bit and isinstance(param, bnb.nn.Params4bit):
-                    quant_storage = self.hf_quantizer.quantization_config.bnb_4bit_quant_storage
-                    # For compatibility with older PT version - see: https://github.com/huggingface/peft/pull/1635
-                    nb_params = (
-                        quant_storage.itemsize if hasattr(quant_storage, "itemsize") else quant_storage.element_size()
-                    )
-                    total_numel.append(param.numel() * 2 * nb_params)
+                    if hasattr(param, "element_size"):
+                        num_bytes = param.element_size()
+                    elif not hasattr(param, "quant_storage"):
+                        num_bytes = 1
+                    else:
+                        num_bytes = param.quant_storage.itemsize
```

Comment:
nit - it's clearer to reference the attr `quant_storage` on the logic check that checks whether or not the object has this attribute 

```suggestion
                    elif hasattr(param, "quant_storage"):
                        num_bytes = param.quant_storage.itemsize
                    else:
                        num_bytes = 1
```